### PR TITLE
Add Matariki manual logic

### PR DESCRIPTION
**Pull request by benaitcheson**
## Description of changes

Added Matariki Public Holiday. Link to definitions PR [here](https://github.com/TandaHQ/definitions/pull/34).

## Notes for code reviewers

There was no way of writing a method for these days so they were hard coded into the file.

### CODE REVIEW

File: lib/generated_definitions/nz.rb [hunk 1]
```diff
@@ -2,7 +2,7 @@
 module Holidays
   # This file is generated by the Ruby Holidays gem.
   #
-  # Definitions loaded: definitions/nz.yaml
+  # Definitions loaded: ../definitions/nz.yaml
```
Comment (calebleungg) [inline lib/generated_definitions/nz.rb:5]: Make sure to remove this line here from our local file trick

Comment (calebleungg) [inline lib/generated_definitions/nz.rb]: Make sure to remove this change here from our local file trick

**Review by calebleungg — CHANGES_REQUESTED** [review]

File: lib/generated_definitions/nz.rb (hunk 1 above)
Comment (benaitcheson) [reply lib/generated_definitions/nz.rb]: Good pick up. Thanks.

**Review by calebleungg — APPROVED** [review]

File: lib/generated_definitions/nz.rb (hunk 1 above)
Comment (calebleungg) [reply lib/generated_definitions/nz.rb]: btw, you'll probably have to change the `payaus` commit hash to reference the one 'on-merge'

**Pull request merged**